print-bootp.c

```diff
@@ -993,6 +997,51 @@ rfc1048_print(netdissect_options *ndo,
 				break;
 			    }
 
+
+			case TAG_SZTP_REDIRECT:
+			    {
+				/* as per https://datatracker.ietf.org/doc/html/rfc8572#section-8.3
+				 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-...-+-+-+-+-+-+-+
+				 |	  uri-length		  |	     URI		  |
+				 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-...-+-+-+-+-+-+-+
+
+				 * uri-length: 2 octets long; specifies the length of the URI data.
+				 * URI: URI of the SZTP bootstrap server.
+				 */
+				u_int suboptnumber = 1;
+
+				while (len >= 2) {
+					suboptlen = GET_BE_U_2(bp);
+					bp += 2;
+					len -= 2;
+					ND_PRINT("\n\t	    ");
+					ND_PRINT("instance#%u: ", suboptnumber);
+					if (suboptlen == 0) {
+						ND_PRINT("[ERROR: suboption length must be non-zero]");
+						bp += len;
+						len = 0;
+						break;
+					}
+					if (len < suboptlen) {
+						ND_PRINT("[ERROR: invalid option]");
+						bp += len;
+						len = 0;
+						break;
+					}
+					ND_PRINT("\"");
+					nd_printjn(ndo, bp, suboptlen);
+					ND_PRINT("\"");
+					ND_PRINT(", length %u", suboptlen);
```

Comment:
DHCP options typically print the length first in any case, and then either print an "invalid" marker or the option value, which makes it possible to see the cause of the problem when the length is invalid. It would be useful to follow the same pattern for the suboptions here.

Reply:
As you mentioned above, this was modeled after the implementation of DHCP_USER_CLASS:
https://github.com/the-tcpdump-group/tcpdump/blob/19c46646c07bc76e403856d18dc0611a96a1627d/print-bootp.c#L985-L988

It sounds like you'd rather see it modeled after AGENT_CIRCUIT's method of printing multiple items?
https://github.com/the-tcpdump-group/tcpdump/blob/19c46646c07bc76e403856d18dc0611a96a1627d/print-bootp.c#L870-L893

Current:
```
           SZTP-Redirect (143), length 141: 
                   instance#1: "https://sztp1.download.com", length 26
                   instance#2: "https://sztp2.download.com:8080", length 31
                   instance#3: "https://10.10.0.0", length 17
                   instance#4: "https://10.10.0.0:8000", length 22
                   instance#5: "https://[2001:4860:4860::8888]:8080", length 35
```

Proposed based on feedback:
```
           SZTP-Redirect (143), length 141: 
                   length 26: "https://sztp1.download.com"
                   length 31: "https://sztp2.download.com:8080"
                   length 17:: "https://10.10.0.0"
                   length 22: "https://10.10.0.0:8000"
                   length 35: "https://[2001:4860:4860::8888]:8080"
                   length 0: ""
                   length 999: length goes past end of option
```

Reply:
Yes, that's the idea, thank you for confirming. This way it would be easier to see what is on the wire, especially if the packet is not well-formed.